File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.java
class Solution {
    public int findMin(int[] nums) {
        if(nums.length==1){
            return nums[0];
        }else if(nums.length==2){
            if(nums[0]>nums[1]){
                return nums[1];
            }else{
                return nums[0];
            }
        }
        int p1 = 0;
        int p2 = nums.length-1;
        while(p2>0){
            if(nums[p1]<nums[p2]){
                return nums[p1];
            }else{
                p1 = p2;
                p2 -= 1;
            }
        }
        return nums[1];
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        if 0 in nums:
            zero_cnt = 0
            for i in nums:
                if i == 0:
                    zero_cnt+=1
            if zero_cnt == 1:
                prods = []
                p = 1
                for i in nums:
                    if i != 0:
                        p*=i
                for i in nums:
                    if i ==0:
                        prods.append(p)
                    else:
                        prods.append(0)
                return prods
            else:
                prods = []
                for i in nums:
                    prods.append(0)
                return prods
        prods = []
        p = 1
        for i in nums:
            p *= i
        for i in nums:
            ans = p//i
            prods.append(ans)
        return prods